### DISCUSSION TITLE
DB

### DISCUSSION THREAD

**Discussion by thorbuckwel** [DavidRieman/WheelMUD#165] [Q&A]
Hello, 

My name is William Buckwell and I'm an adjunct instructor teaching C#. In my class I have my students create a text adventure but I have had a hard time showing them an example in C#. I ran across your "MUD" and would love to show it to them as an example framework and what all goes into it. I would like to show them a running example however, when I go to build it I get a little confused on how and where the databases go. Is there a setup that shows where these go?

Further information about me is I graduated with a 2-year degree but have never been in real-world programming so I knew a little beyond text books such as "starting out with C#" by Gaddis.

**Comment by DavidRieman** [comment]
Hi,

Yeah, finding the windows local data area for an app is not straight forward, so the app creates a shortcut to it when you run the server.  For example, you may find in your folder like:
`GIT\WheelMUD\binDebug\netcoreapp3.1`
the shortcut file `_WheelMUD_DataFolder`
which points to your data folder, like:
`C:\Users\username\AppData\Roaming\WheelMUD\WheelMUD`

However, if you'd like to explore the DB, we recently abandoned the SQLite DB so you'll want to explore the RavenDB store instead, for which you don't need to locate the local data either.  You can explore that by entering a command from the server console too while it is running:  `DB`
That opens a browser for which you can handily explore your embedded RavenDB instance.  E.G. you can click the "Documents" tab at the top left to start exploring the stored entities, like the rooms.

Sorry for the late reply.  I do need to update the "getting started" resources to cover this stuff accurately again.  We had some of this stuff covered ages ago, but it changed w/the evolution of how we do storage to the far more flexible/powerful document DB approach, and I haven't spent enough time on updating the docs for this.
